back/src/chat/channel/channel.controller.ts

```diff
@@ -34,62 +37,118 @@ export class ChannelController {
 		return await this.channelService.getAllChannelConnections();
 	}
 
-	@Get(':id/connection')
+	@Patch(':targetChannelId')
+	@UseGuards(...AuthenticatedGuard, IsOwnerGuard)
+	@UsePipes(ValidationPipe)
+	async updateChannel(
+		@Req() req: Request,
+		@Param('targetChannelId', ParseIntPipe) channelId: number,
+		@Body() data: ChannelDto
+	): Promise<Channel> {
+		return await this.channelService.updateChannel(req.user.id, channelId, data);
+	}
+
+	@Get(':targetChannelId/connection')
 	@UseGuards(...AuthenticatedGuard, IsInChannelGuard)
-	async getChannelConnection(@Param('id', ParseIntPipe) channelId: number): Promise<ChannelConnection[]> {
+	async getChannelConnection(@Param('targetChannelId', ParseIntPipe) channelId: number): Promise<ChannelConnection[]> {
 		return await this.channelService.getChannelConnections(channelId);
 	}
 
-	@Post(':id/join')
+	@Post(':targetChannelId/join')
 	@UseGuards(...AuthenticatedGuard)
-	@UsePipes(new ValidationPipe())
-	async join(@Param('id', ParseIntPipe) channelId: number, @Req() req: Request, password: string) {
+	@UsePipes(ValidationPipe)
+	async join(@Param('targetChannelId', ParseIntPipe) channelId: number, @Req() req: Request, password: string) {
 		return await this.channelService.join(req.user, channelId, password);
 	}
 
-	@Post(':id/quit')
+	@Post(':targetChannelId/quit')
 	@UseGuards(...AuthenticatedGuard)
-	@UsePipes(new ValidationPipe())
-	async quit(@Param('id', ParseIntPipe) channelId: number, @Req() req: Request) {
+	@UsePipes(ValidationPipe)
+	async quit(@Param('targetChannelId', ParseIntPipe) channelId: number, @Req() req: Request) {
 		return await this.channelService.quit(req.user, channelId);
 	}
 
-	@Post(':id/invite/:userid')
+	@Post(':targetChannelId/invite/:targetUserId')
 	@UseGuards(...AuthenticatedGuard)
```

Comment:
Can't you use `isAdminGuard` on this route?
Thus avoiding the check on the [channelService](https://github.com/42rave/transcendence/blob/7ca3341ad5fa4dfac5c404c8d338d7a26e106c8a/back/src/chat/channel/channel.service.ts#L308-L312)

Reply:
Indeed but I think I liked the idea of being able to tell that the reason you cannot invite someone in the channel is because the channel does not exist, not because you are not admin.
What do you think? Maybe @Elena-Lou or @Deacllock could share their thoughts?

Quick recap:
To invite someone in the channel (add them on the invite list) you need to be admin.
There is a guard (IsAdminGuard) that checks this condition and returns true or false therefore allowing or rejecting the request on the route if the conditions are met/not met.

Using this guard here would still mean one database query, less code inside the function (so cleaner code) but would not differentiate if the guard action comes from a non existing channel or from a user not admin of said channel.

Reply:
You can throw an error on the guard:

```ts
async canActivate(context: ExecutionContext): Promise<boolean> {
  [...]
  if (!await this.channelService.isUserAdminInChannel(targetUserId, targetChannelId))
    throw new ForbiddenException('You are not admin');
  return true;
}
```

The default guard behavior is:
- If the canActivate return true nothing happens
- else it throw something like BadRequest or Forbidden exception (don't remember exactly sorry).

But you can if you want throw a custom error depending on a condition, else return true!

Reply:
Ok gotcha.
But then I end up doing two requests in the guard for :
- "does the channel exist ?" 
- "is the user admin ?"

Reply:
Yeah i guess..
Perhaps the best thing to do would be to implement middleware to retrieve the targetChannelId if given in the parameters, and inject it into the request context, so that the guards only have to execute an if statement instead of fetching the channel from the database twice.
But honestly, fetching the channel twice is acceptable, as long as the use of routes using these guards isn't really prone to spam.

Reply:
I guess, fetching the channel twice it's less effort than implementing a new middleware etc.